### PR TITLE
Add conditional target architecture when getting supported versions

### PR DESCRIPTION
**Pull request by williamhammond**
# Reason for This PR

Fixes https://github.com/firecracker-microvm/firecracker/issues/2467

## Description of Changes

Conditionally excludes v0.23 from version mapping for aarch64.  

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] Any newly added `unsafe` code is properly documented.
- [ ] Any API changes are reflected in `firecracker/swagger.yaml`.
- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

### CODE REVIEW

**Comment by williamhammond** [comment]
Looks like the failure wasn't due to the change. It happened in the perf tests

**Comment by gc-plp** [comment]
@williamhammond thanks for the PR! I retriggered the build to check if the failure is consistent.

**Review by sandreim — COMMENTED** [review]
Change looks good. Can you also add a negative integration test for aarch64 that validates one cannot ever save a snapshot which targets v0.23.0 ?

**Comment by williamhammond** [comment]
Darn Diana beat me to the tests :smile:, Is there anything else i need to do or wait for approval?

**Comment by williamhammond** [comment]
Actually now that we are preventing 0.23 from getting pulled in the version map, can the additional version guards in https://github.com/firecracker-microvm/firecracker/blob/646ec96733d6aa72586e69dc9bcd0b38dab2df7d/src/vmm/src/persist.rs#L270 be removed?

**Comment by dianpopa** [comment]
> Darn Diana beat me to the tests 😄, Is there anything else i need to do or wait for approval?

I do not think there is anything else left to do, just waiting for approval. Wanted to thank you for your PR and involvement! 👍  Feel free to contribute more!

File: tests/integration_tests/functional/test_snapshot_version.py
Comment (sandreim) [inline tests/integration_tests/functional/test_snapshot_version.py:41]: I would rename it something more specific like `test_create_v0_23_snapshot`.

File: tests/integration_tests/functional/test_snapshot_version.py
Comment (sandreim) [inline tests/integration_tests/functional/test_snapshot_version.py:42]: Please also update short description.

File: tests/integration_tests/functional/test_snapshot_version.py
Comment (sandreim) [inline tests/integration_tests/functional/test_snapshot_version.py:59]:
```suggestion
    test_microvm.pause_to_snapshot(
        mem_file_path="/snapshot/vm.mem",
        snapshot_path="/snapshot/vm.vmstate",
        version="0.23.0")
```

File: tests/integration_tests/functional/test_snapshot_version.py
Comment (sandreim) [inline tests/integration_tests/functional/test_snapshot_version.py:61]: Since this test always passes on x86_64 I suggest a diff snapshot so it runs faster.

File: tests/integration_tests/functional/test_snapshot_version.py
Comment (dianpopa) [reply tests/integration_tests/functional/test_snapshot_version.py]: Done.

File: tests/integration_tests/functional/test_snapshot_version.py
Comment (dianpopa) [reply tests/integration_tests/functional/test_snapshot_version.py]: Done.

File: tests/integration_tests/functional/test_snapshot_version.py
Comment (dianpopa) [reply tests/integration_tests/functional/test_snapshot_version.py:62]: Ok

**Review by dianpopa — COMMENTED** [review]
@sandreim addressed your comments. PTAL!

**Review by alindima — APPROVED** [review]
LGTM

**Review by sandreim — APPROVED** [review]

**Pull request merged**
